management/server/account.go

```diff
@@ -1888,6 +1890,8 @@ func (am *DefaultAccountManager) SyncAndMarkPeer(ctx context.Context, peerPubKey
 }
 
 func (am *DefaultAccountManager) CancelPeerRoutines(ctx context.Context, peer *nbpeer.Peer) error {
+	peerUnlock := am.Store.AcquireWriteLockByUID(ctx, peer.Key)
```

Comment:
We should move the peerLock after the account lock otherwise we might still experience overwrites if the peerLock is called while an accountWrite lock is currently active